src/components/Home/Hero/ExchangeWidget/CoinInput/CoinInput.js

```diff
@@ -70,17 +79,19 @@ class CoinInput extends PureComponent {
     }
   };
 
-  UNSAFE_componentWillReceiveProps = nextProps => {
-    if (nextProps.price.fetching !== this.state.fetching) {
-      this.setState({ fetching: nextProps.price.fetching });
+  componentDidUpdate = prevProps =>  {
+    if (this.props.price.fetching !== this.state.fetching) {
+      this.setState({ fetching: this.props.price.fetching });
     }
 
-    if (nextProps.lastEdited !== nextProps.type || !this.state.value || this.state.value === '...') {
```

Comment:
@pragai I am not sure what did you do here, however the lastEdited pop is important becuase we use it to 'fixate' the amount the user gets (in the rate case that the price changes in the exact instance the person places an order).

Please let me know

Reply:
The main reason of the bug this fix addresses was that in the original code the lastEdited property was linked to the selected coin. The base value was "deposit" and it was not changed when user edited the "received" amount.

const mapStateToProps = ({ selectedCoin, price }) => ({ selectedCoin, price, lastEdited: selectedCoin.lastSelected });

So, in the UNSAFE_componentWillReceiveProps method when this was checked, the "deposit" value was always the same until user selected a new coin on receive side but then "receive" was always on.

For a start I replaced UNSAFE_componentWillReceiveProps with componentDidUpdate to have a clear page on this. It seemed then that checking lastEdited is not necessary in the end as amount_base and amount_quote are also received in the ajax data.

So if I'm correct the lastEdited prop would be used to enable changing the deposit or receive values in case of a new price comes in. And it would change the one which was not edited by the user. Am I right? Then I put this into the new code, too.